GeneWeb v7.1beta + templates v7.1 beta

hgouraud opened the pull request:
cgl=on was not handled
confusion between im and images, sp and spouses

hgouraud committed:
Produce cache files for names and places

hgouraud committed:
Fix spouses, images and cgl for m=RL

hgouraud committed:
maintain im/image, sp/spouse 0/1 compatibility

hgouraud committed:
Suppress test for sp/im=0/1 in ml code

hgouraud committed:
fix m=C without parents. Return -1 if list empty

hgouraud committed:
Change birthday button color. Add year and age where needed

hgouraud committed:
Fix expected int, found  in cousmenu.txt

hgouraud committed:
fix cousins (index out of bounds)

swrup commented:
I converted it to draft, there is some TODO left in the new code

hgouraud commented:
Fixed TODO check for Sys.max_array_length.
The other TODO is a comment by you! : 
  (* TODO type for this tuple?; why list of level? *)
Because of implex, a person may be a "cousin" at different levels.

hgouraud committed:
Add failwith if cousins array is too big

hgouraud committed:
move min_max_date to Cousins

hgouraud committed:
fix im=off/0 and suppress max_anc/desc_lev_default

hgouraud committed:
e/b im/sp/ma

*missing im=0 in dagDisplay.ml
*m=TV as default
*missing <br> in tree
*fix tree m=D&t=V

This reverts commit dcc62bd265ba764cff06ac40a5cb3b4252e3dc77.

swrup commented on lib/cousins.ml:
```diff
@@ -246,40 +252,65 @@ let init_cousins_cnt base max_a_l max_d_l p =
+        if
+          i + 1 > Array.length cousins_dates
+          || j + 1 > Array.length cousins_dates.(i)
+        then (-1, -1)
```
Shouldn't this be None too?

swrup commented on lib/util.ml:
```diff
@@ -759,6 +759,8 @@ let max_ancestor_level conf base ip max_lvl =
   loop 0 ip;
   !x
 
+let max_descendant_level _conf _base _ip _max_lvl = 120
```
maybe this should be a constant and be in Cousins?

hgouraud replied:
I did some cleanup, but a check-up pass on those max values is needed

swrup commented on lib/perso.ml:
```diff
@@ -2615,141 +2603,73 @@ and eval_person_field_var conf base env ((p, p_auth) as ep) loc = function
-        | _ -> max_desc_level_default
-      in
-      let l = Cousins.cousins_l1_l2_aux base max_a_l max_d_l l1 l2 p in
+      let l = Cousins.cousins_l1_l2_aux conf base l1 l2 p in
```
I mean not visible outside cousins.ml (not in cousins.mli)

hgouraud committed:
some cleanup for max_anc/desc_level and 120 ...

a2line committed:
lexcheck

michelmno commented:
Something strange, above comment says that branch has no conflict with geneweb:master,
but if I try locally to rebase [hgouraud:fix-rl](https://github.com/hgouraud/geneweb/tree/fix-rl) on [geneweb:master](https://github.com/geneweb/geneweb/tree/master) then I have conflict related to 
cid 858a88f540 present in geneweb:master but not in hgouraud:fix-rl 
Is there something I do not understand ?

hgouraud commented:
I dont understand myself what is happening with this branch. I tried to fix it in several ways to no avail!!
I just tried again! Seems better now.

a2line committed:
Use same function in forms to show/hide link/create inputs

Fix the design and accessibility of new inputs for witnesses/fwitnesses.
The adequate jQuery selector allows to use the same function to show/hide
inputs depending of link/create of witnesses/relations (updind) and
parents/children/fwitnesses (updfam).
When “link”ing show the sex and a link to mod_ind link to individuals.
Remove last XHTML %/ and extra-spaces in parameter of inputs.
Use some Bootstrap custom checkboxes.

a2line committed:
better shortcut typography

a2line committed:
latin small letter alpha u+0251 ?

a2line committed:
revert rename with space

a2line committed:
clean_comment_tagS

fix #1532

a2line committed:
factorise male/female line, support cancel_links

fix typo s/cell.father/cell.family/ (#1531)

a2line committed:
switch to b.full_siblings=no, remove circular link on self for f2

* full_siblings is default, add bvar full_siblings=no to remove self from f1-f4
* rename evar e/bvar b, clean conditions

a2line committed:
typo b.defauLt_lang

a2line committed:
all parameters active by default with shorten variables

* use e/b, im=0/sp=0/half=0
* remove toggle icons, use primary button with a normal size
* use %url_set when possible instead of %foreach;bindings
* remove unused b.default_fnav_images/semi/spouse

a2line committed:
remove capitalisation

a2line committed:
use ampersand for URL parameters

a2line committed:
revert access to m=SND_IMG_C on menubar

a2line commented:
This branch needs more testing it's not good at the moment. Consider it's draft/wip.

hgouraud committed:
do not insert & after ? in url (%set.url...)

hgouraud committed:
add explicit Incorrect request title and 'error #n' message in request.ml

hgouraud committed:
add 1 to array size for cousins

a2line committed:
add three missing empty cells if self is single

hgouraud committed:
replace html comments by empty string

hgouraud committed:
correct another url_set2 in copyr.txt

hgouraud committed:
fix erronous sosa number on top left in compact tree

fix #1537

hgouraud committed:
missing parents on h7

partialy fix #1539

hgouraud committed:
fix m=PPS pb when place starts with (

hgouraud committed:
add new carrousel actions, suppress duplicate sn cass

hgouraud committed:
suppress trailing backslash

hgouraud committed:
compute %cous_implx_cnt; only if e.c_implx=on or 1. return null otherwize

hgouraud committed:
use new url_set2; suppress lang= if default_lang

hgouraud committed:
cleanup language choice for templm

a2line committed:
highlight default_lang in datalist, show “default” in tooltip

hgouraud committed:
missing test for sp in descendDisplay.ml

a2line committed:
show periods of relation again fixing global alignment

a2line committed:
fix some buttons and rendering problems

a2line committed:
revert “rename pvar to p_of_index”

* fix title m=A&t=D/X
* remove unused functions in ancmenu

a2line committed:
port blindly updRLM changes to updDAG

hgouraud committed:
fix last marriage. Manage & and mariage_date. Deal with multi-marriages
(wip)

a2line committed:
add class on form, remove table

* remove [] around name of notes

a2line committed:
fix home buttons alignment, HTML head

a2line committed:
nesting span is valid HTML

a2line committed:
revert opacity-0 on navbar, add a CSS scroll offset

* with a scroll-padding-top that equals the height of the navbar the anchors going up in the page are not hidden by the navbar itself

hgouraud committed:
fix a %sbrother ...; in %url_set;, put new evar at end of url

hgouraud committed:
manage d['|e ]%s; add vowels list in .gwf file

michelmno committed:
Limit nb of generations for father/mother buttons

When depressing father or mother buttons in Descendant Tree view
we should not increase v=xx value more than limited value defined by
max_desc_tree and max_desc_level variables in base.gwf

Note the usage of e.m test in place of dtree for button titles
to display correct one for t=V and not only for t=V and T=TV.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

a2line committed:
s/spouse.xxx/xxx.spouse/

michelmno committed:
Change color of spouses info between t=H/I for descendant table view

color to be red when info not currently displayed (similar to other icons)

Question: why icon is in two parts, a 'i' character and a couple ?
why not to simplify to only one of the two ?

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
add %witness.index; %witness.sex; %fwitness.index;

* occ returns 0 rather than empty string

a2line committed:
Bump jQuery to v3.7.1

a2line committed:
remove intrusive generations title in CSV

michelmno commented on hd/etc/menubar.txt:
```diff
@@ -342,11 +342,8 @@ fr: Sélectionnez les modules en cliquant sur le bouton correspondant (max 15).
-                <a class="dropdown-item" href="%prefix;%access;&m=D&t=T&v=4"%if;(not has_parents) %laY;%end;><span class="fa fa-sitemap fa-fw mr-2"></span>[*descendants tree]</a>
-                %)
-                <a class="dropdown-item" href="%prefix;%access;&m=D&t=D&v=%max_desc_level;"><span class="fa fa-code-branch fa-rotate-90 fa-fw mr-2"></span>[*descendant tree view]</a>
+                <a class="dropdown-item" href="%prefix;%access;&m=D&t=TV&%if;(e.v!="")v=%e.v;%else;v=3%end;&sp=1&ma=1"%if;(not has_parents) %laY;%end;><span class="fa fa-sitemap fa-fw mr-2"></span>[*compact descendants tree]</a>
```
@a2line , I strongly dislike the fact you pulled commit I signed with changes (in summary and code changes)  without any previous discussions in related PR (https://github.com/hgouraud/geneweb/pull/7).  Not a good practice.

a2line replied:
https://fr.wiktionary.org/wiki/psychorigidité 2.0

hgouraud commented:
some problems remain on this branch (ml related).
I wont be able to address them before tuesday sept 5th

a2line commented:
some heavy problems… but less than master…

a2line committed:
fix alignment of single individuals

hgouraud committed:
New Hutil.interp_no_env for buttons_rel

* Systematic treatment of im, sp, ma.
* Add missing & after %url_set.
* Add buttons_rel in relationDisplay.
* Fix link_next/next_txt.

swrup commented on lib/cousinsDisplay.ml:
```diff
@@ -25,6 +25,16 @@ let cnt = ref 0
+    match (Util.p_getenv conf.env "sp", Util.p_getenv conf.env "spouse") with
+    | Some ("off" | "0"), _ | _, Some "off" -> false
+    | _, _ -> true
+  in
```
maybe we should make it a function in Util so we can have something like: 
 ```ocaml
if Util.is_on conf.env ["sp";"spouse"] then ... else ...
```
Or even better, use types instead of strings so we can have
 ```ocaml
if Util.is_on conf.env Spouse then ... else ...
```

swrup replied:
If you want to use "sp" instead, why not just changing every "spouse" by "sp" in template code and .ml (well this will break others people template but...)

And why not just keep "spouse"?

I think having two string for the same thing is bound to cause a bug

a2line committed:
remove unused functions and file

a2line committed:
(wip) buttons

swrup commented on lib/cousinsDisplay.ml:
```diff
@@ -37,17 +47,13 @@ let give_access conf base ~cnt_sp ia_asex p1 b1 p2 b2 =
+              :> Adef.escaped_string)
+        ^^^ "&bd="
+        ^<^ (Option.value ~default:(Adef.encoded "0")
+               (List.assoc_opt "bd" conf.env)
```
if you manage to rewrite this string concatenation as a single `Printf.sprintf` it would be nice

swrup commented on lib/image.ml:
```diff
@@ -318,7 +318,11 @@ let carrousel_file_path conf base p fname old =
+    String.concat Filename.dir_sep [ carrousel_folder conf; dir; fname ]
+  in
+  if f.[String.length f - 1] = '\\' then String.sub f 0 (String.length f - 1)
+  else f
```
this is to remove last directory separator (only on windows?) if fname was empty?

maybe it is clearer to not have fname in the list we apply `String.concat` to if it is empty
```ocaml
  let f =
    String.concat Filename.dir_sep ([ carrousel_folder conf; dir] ) @ ( if fname = "" then [] else [fname] ) )
  in
```

hgouraud replied:
done

hgouraud committed:
get max_a_l and max_d_l from base_env

hgouraud committed:
person_index in dagDisplay, better mngt of i/ei, rework URLs in cousinDisplay

hgouraud commented on lib/cousinsDisplay.ml:
```diff
@@ -414,9 +416,9 @@ let print_anniv conf base p dead_people level =
-      ^>^ "&spouse=on"
+      ^<^
+      if spouse = None then "&" ^<^ acces_n conf base (Adef.escaped "4") c
+      else Adef.escaped ""
```
At this stage, I did not try to factorize

hgouraud committed:
better management of trailing slashes \

hgouraud committed:
comments on lang treatment in url_set_aux

hgouraud committed:
suppress && for m=A&t=T. Correct im=0

hgouraud committed:
fix count and maxv error

fixes #1558 and #1559

hgouraud committed:
set family_cnt to 0 if None

a2line committed:
show q_time with colours on a new icon, smaller home icons on left for trees

If debug is on, we add a fourth hourglass icon on home buttons that is working
as a traffic light (green/orange/red) depending on CPU loading time of the page.
Switch home buttons to the left with CSS on various trees to avoid overlapping.

a2line committed:
wip

a2line committed:
fix negative number error

todo : fix extra rows
use <> instead of decimal html entities

a2line committed:
missing titles on trees (wip)

a2line committed:
add “old” for ages in english

fixes #1279 (age/age at death indications already on tooltip)

hgouraud committed:
correct templm for asc/desc headings

hgouraud committed:
fill in arbre_hi_indexes; fix 1558

a2line committed:
remove debug, add row removal on headers

* remove forgotten cousin.l1.l2 counts used to debug m=C
* add stretched-links on table headers to remove column on fly

a2line committed:
revert partialy “manage nb of desc. at gen”

hgouraud committed:
Do not increment visit counters if m=IM, IM_C, SRC and DOC

hgouraud committed:
correct/update util_test (vowels)

hgouraud committed:
tesst has_parents before father/mother (fix 1561)

hgouraud committed:
inverted im status in destree.txt

hgouraud committed:
Revert "fill in arbre_hi_indexes; fix 1558"

This reverts commit 547db7970ccf505572c143fd1fdc0bb578d0c986.

hgouraud committed:
better solution for arbre_hi_indexes nth_0 returns 0 if empty

hgouraud committed:
explicit only.txt location. Add a note on -only file launch param

hgouraud committed:
translate title for compact tree

hgouraud committed:
typo in dag.txt title

hgouraud committed:
fix a1 module as per issue 1338

a2line committed:
add a bottom padding of the size of the navbar

hgouraud committed:
reformat print_sstrings; create get_opt function for ma, sp, im

hgouraud committed:
use get_opt for im, ma, sp; cleanup multiple print_sstring

hgouraud committed:
correction for d[e |']; cleanup titles in dag.txt; some experiments with ::|xxx|

hgouraud committed:
some experiments with ::|xxx|

hgouraud committed:
fix issue 1556 (arbre_hi.txt)

hgouraud committed:
arbre_hi: correct missing sosa display

hgouraud committed:
set version name to 7.1-beta

hgouraud committed:
fix log error message; New version of arbre_hi_indexec

michelmno committed:
Remove ref to not accessible favicon.ico

Solution for issue #1574

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Avoid empty plugins list in welcome page

as reported in issue #1576

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Ignore failure on connection shutdown request

to avoid regular messages logged by gwd,
as reported by issue #1517

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
invert graph/graphlinks in arbre_hi.txt; correct missing lines for v>6;
correct undefined spouse.has_parents"

michelmno committed:
Avoid Erreur d analyse de la valeur pour line-height

as reported by browser console, when display descendant compact tree.

Full reported browser console error:
07:21:50,225 Erreur d’analyse de la valeur pour « line-height ».  Déclaration abandonnée.

I identified source of the problem in dag.txt as a missing else leg
in the code change of commit b9fbd74 "a few new features for templates"

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
fix issue 1580. Missing else clause

hgouraud committed:
fix issue 1578. Branch by name title

hgouraud committed:
cleanup arbre_hi_tools.txt. remove crawling nz, pz, ocz

a2line committed:
various h-trees cleanup

Update depreciated js .reload() since jquery v3.
Move jquery/jquery.line in js.txt, call them before h-trees
Ugly temporary fix for jquery.line impleX positions
Hard revamp of tools==buttons (still needs to be improved for icons/transl.)

a2line committed:
factorise headers of tables

a2line committed:
print hz tree without changing Sosa1 (wip, to verify)

a2line committed:
cleaner indent

a2line committed:
use url_set

a2line committed:
minor fixes

a2line committed:
add padding-bottom to compensate fixed navbar

a2line committed:
missing space

hgouraud committed:
fix index out of bounds in cousins.ml

hgouraud committed:
report nb errors in qery_time

a2line committed:
revert bp-6, fix ) in title (to squash)

a2line committed:
add margin-bottom to compensate fixed navbar

hgouraud committed:
show spouses of last line in compact tree

hgouraud committed:
repeated dates in short display person

hgouraud committed:
more fix to issue #1582 (anclist.txt)

hgouraud committed:
generalize url_set.a_b_c for multiple evar. Rename url into url_in_env in perso.ml

hgouraud committed:
correct carrousel after renaming url to url_in_env

hgouraud committed:
fix issue #1587 (multiple lang=en)

hgouraud committed:
always compute and display query_time

hgouraud committed:
in dag.txt provide person_index.1.surname/first_name; Fix buttons when not interpreted by template

hgouraud committed:
define url_set_new.new_url

hgouraud committed:
treat default lang when reconstructing url

hgouraud committed:
title for m=RL; fix index-out-of-bounds in init_cousins

a2line committed:
add an intrusive bug icon when errors are founds

Note that the counter detects errors that are not shown in logs!

hgouraud committed:
%prefix_new_iz.iz;, %prefix_new_pz.pz.nz.ocz; %prefix_new_ix.index;

a2line committed:
support 1 generation

a2line committed:
fix desc. tree url, remove comment

a2line committed:
fix broken tools on templm

Use min() to define max level. Self now at v=v1=0.
Table and partial table are now t=X/Y (not D).

hgouraud committed:
replace %prefix_new_iz; by %set_url_pz.iz; %prefix_new_pz.pz.nz.ocz; by %set_url_pz.pz.nz.ocz;

a2line committed:
use new url_set_iz/pnocz

hgouraud committed:
disable runtime_warnings if not -debug

hgouraud committed:
Restore Missing fn and sn message if empty search

hgouraud committed:
reorder evar

hgouraud committed:
fix issue 1599, handle multiple translate as in [aa::|[xx]:[yy]|]

michelmno commented:
@hgouraud is this branch ready to be merged in geneweb:master branch ?
@Swrup , are you ok to merge it ?

a2line committed:
simpler lists, icons for buttons, no header

a2line committed:
use a non-breaking space, remove determiner from title

a2line committed:
check if HTML exists

a2line committed:
use non-breaking spaces before counts

a2line committed:
fix second button group horizontal alignement

a2line committed:
fix upto link (m=X), simplify HTML

hgouraud committed:
rework print_cousins to use [cousins.v1.v2]

a2line committed:
revert ground state to restore fan centering

hgouraud committed:
report undefined vars in title of qtime error

a2line committed:
update forms, fix #rows, todo: fix search!

a2line committed:
fix home buttons to top-left

add home to header and start remove link_to_welcome (a lot still left)
fix unbound var (ancsosa not_are_married/updfam readonly)
fix margin for home on left

a2line committed:
fix icons for guests

a2line committed:
improve formatting and buttons, fix sort by date

hgouraud committed:
textual dates badly transported in a gwu to gwc cycle

Fix #1602

hgouraud committed:
some update to install_cgi.sh

a2line committed:
new bvar.book_max_results limits book results

* new .gwf variable book_max_results (default is 1000 if empty)
* restore wizard alert on top
* remove disabled from hyperlink, use div instead
* replace table by flex and adapt overall alignments
* fix #1603 (Firefox bad width)

a2line committed:
add !important needed for Firefox

a2line committed:
remove home from non-template pages with menubar

home was printed in double

hgouraud committed:
add chld of, spouse of to person when merging (mergeDisplay.ml)

hgouraud committed:
fix %s in relationship translations

issue #1591

hgouraud committed:
use place_v7 instead of place

old place suppressed, place_v7 renamed place

Commit:
use new key and new ini upon data update

* missing nbsp before unit
* s/50000/50000
* add ids for anchor on entries

a2line committed:
CSS scroll fixes specific to books/tables

a2line committed:
update CHANGES for release 7.1

a2line committed:
enhance the pyramids of population

Higher bars and larger padding on central column.
Fix missing labels, add two fields “interval” and “limit” to the form
with translations for them and a submit button.
Limits: interval < 130 and limit < nb_of_living_persons.

a2line committed:
add home buttons on statistics

a2line committed:
fix sex for empty parent/addfam cases

a2line committed:
a lot of various W3C fixes

quickfix for failing functions on welcome #1607

hgouraud committed:
add hidden e.lang, missing &

fix #1609 and mayby #1610

michelmno committed:
Add few missing bvars in a.gwf (step1)

* Add a notice for next [WIP] about 'boolean' on top of file.

* Add for each some prefix comment lines.
* for 'boolean' specify non-default as commented line.
* for numeric directly specify default value as commented line.

This is a step1 commit, because just started to investigate
what is missing in a.gwf.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Comment out max_anc* max_desc* bvars in a.gwf

because those variables have default values
so not mandatory if do not want to change those default.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

a2line committed:
fix margins for cgi login

a2line committed:
missing maximize-height for CDN

but select still buged on ancmenu using CDN

michelmno committed:
Replace %bvar.static_path; by %static_path; (step2)

* This is a complement of cid f4c4e1b7c
  the next fanchart.txt added later by cid b9fbd7492
  missed this change.
* Add also related comment in CHANGES file for 7.0.1

* in same fanchart.txt update all.min.css version ref from 6.2.1 to 6.4.0
  as already done in other place with cid 85ccf8106

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
use default_allowed_tags if tags.txt absent. report Warning

hgouraud committed:
define %dummy in chronologie.txt

a2line committed:
hide_querytime_bugs=yes removes also javascript

a2line committed:
spouses' dates missing for visitors

hgouraud committed:
manage undefined variable message for set_var/get_var

hgouraud committed:
rework error reporting (two lists)

a2line committed:
just a few missing semi-columns

a2line committed:
just a few semi-columns replacements

all made by hand with heart but without head, could be misktakes, some urls still probably miss some more ampersands specialy in templm

hgouraud committed:
add more tests in test_sosa.ml

hgouraud committed:
manage get_var.xxx

a2line committed:
rewrite and reorder a.gwf

a2line committed:
only line feed allowed after %include;…

a2line committed:
fix very hard to find %? variable bug!!

Clean some conditions and switch the 4 table_row (function of 11 variables!) to
“long form”; removes a probable parsing bug with “parentheses form”. The traces
are left before the %apply calls in comments because it was very hard (!!) and
it could be usefull for the following. The hyperlink to go back to parents is
now added on every descendants and not only on the elder of each siblings. Add
eight more pixels on margin-left of the container (pl-4 instead of default 3)
to support home on left.

hgouraud committed:
correct married_to in deslist.txt

hgouraud committed:
force home to the left, fix stray space in undefined get_var variable

hgouraud committed:
add conf.image_prefix in birthDeathDisplay. Use /10/10/10 rather than /1000

swrup commented:
ocamlformat version should stay specified

this PR is already huge, format change should come in another one
and preferably after we rebase other bigs PR...

swrup commented on bin/gwd/gwd.ml:
```diff
@@ -1052,7 +1052,19 @@ let authorization from_addr request base_env passwd access_type utm base_file
         basic_authorization from_addr request base_env passwd access_type utm
           base_file command
 
+let explode s =
```
maybe `string_to_char_list` is better name

swrup commented on bin/gwd/gwd.ml:
```diff
@@ -1052,7 +1052,19 @@ let authorization from_addr request base_env passwd access_type utm base_file
+let explode s =
+  let rec exp i l =
+    if i < 0 then l else exp (i - 1) (s.[i] :: l) in
+  exp (String.length s - 1) []
```
you could use String.fold_right or fold_left instead, but this is fine

swrup commented on bin/gwd/request.ml:
```diff
@@ -176,7 +176,7 @@ let very_unknown conf _ =
       Hutil.print_link_to_welcome conf false;
       Hutil.trailer conf
-    | None -> incorrect_request conf
+    | None -> Hutil.incorrect_request conf ~comment:"error #1"
```
thanks, this is nice for debug

swrup commented on bin/gwdiff/gwdiff.ml:
```diff
@@ -135,13 +135,13 @@ let dmy_to_sdn_range_l dmy =
+                 if dmy.month = 12 then 1 else dmy.month + 1
+               else dmy.month);
             day = (if dmy.day = 0 then 1 else dmy.day);
             prec = (if dmy.month = 0 || dmy.day = 0 then Before else Sure);
```
I don't want to look futher into what this code does for now but we had issues with dates and their encoding with day=0 for unknown day.
So the buggyness of this code looks high to me, maybe it should use a function from the Date module?

Also maybe gwdiff should be in geneweb-contrib?

hgouraud replied:
not my code!!

swrup replied:
I know!!
but it was touched because of the format

let's not worry about this code at all

swrup commented on bin/gwu/gwuLib.ml:
```diff
@@ -948,34 +950,34 @@ let print_notes_for_person opts base gen p =
+                   | Epers_Death -> "death"
+                   | Epers_Burial -> "burial"
+                   | Epers_Cremation -> "creamation"
+                   | _ -> ""
```
this match should be a function `epers_name_to_string` defined elsewhere
(also notice the typo on cremation fixed in my gwu fix PR)

hgouraud replied:
ok

swrup commented on etc/LISEZMOI.txt:
thanks, spaces in filenames are evil!

swrup commented on etc/install-cgi.sh:
```diff
@@ -53,28 +55,60 @@ echo "Bases: $BASES"
+# This is a test for cgi commands
+# This should display Lenna:
+# <image of Lenna.jpg>
+# End of test 
```
this comment use your specific setup and account/computer name

hgouraud replied:
ok

swrup commented on lib/GWPARAM.ml:
```diff
@@ -4,6 +4,11 @@
+let nb_errors = ref 0
+let errors_undef = ref []
+let errors_other = ref []
+let set_vars = ref []
```
are those really helpful and needed?

unless I'm mistaken, because of the fork mechanism of geneweb this will be specific to only one connection

maybe you can compute nb_errors from the other ref when needed to be sure you have the right count and not have to worry about it everywhere.

maybe we can update a list of errors for each variant of syslog_level in `syslog` so every error is saved in those.

hgouraud replied:
we have found this count and report very usefull to chase bugs in the templates. The principle being agreed upon, one could certainly improve.

swrup commented on lib/birthdayDisplay.ml:
```diff
@@ -142,17 +150,20 @@ let print_anniversary_list conf base dead_people dt liste =
                 Output.print_sstring conf " ";
-                Output.print_sstring conf (transl conf "years old"));
-            Output.print_sstring conf "</em>"
+                Output.print_sstring conf (transl conf "years old"))
```
this code looks similar too code line 29, it would be nice if it can be factorized

hgouraud replied:
done

swrup commented on lib/cousins.ml:
```diff
@@ -6,13 +6,10 @@ open Util
 let default_max_cnt = 2000
 
-let max_cousin_level conf base p =
+let max_cousin_level conf _base _p =
```
if they _base and _p are unused maybe remove them

swrup commented on lib/cousins.ml:
```diff
@@ -202,20 +235,28 @@ let descendants base cousins_cnt i j =
         let () = load_ascends_array base in
         let () = load_couples_array base in
-        let cousins_cnt = Array.make_matrix (max_l2 + 1) (max_l2 + 1) [] in
+        (* +3: there may be more descendants for cousins than my own *)
```
the +3 looks like a hack
why is it needed?
do the code still works with cousins that have more than +3 descendants level than you?

hgouraud commented on bin/gwu/gwu.ml:
```diff
@@ -53,10 +53,10 @@ let main () =
-       ());
+         let () = Gwdb.load_couples_array base in
+         let () = Gwdb.load_unions_array base in
+         let () = Gwdb.load_descends_array base in
```
ok

swrup commented on lib/searchName.ml:
```diff
@@ -207,4 +207,4 @@ let print conf base specify unknown =
       in
       search conf base sn order specify unknown
-  | None, None -> Hutil.incorrect_request conf
+  | None, None -> Hutil.incorrect_request conf ~comment:"Missing fn and sn"
```
since this will be a error the user will see, maybe we should write first name and surname in full for less confusion

hgouraud commented on lib/advSearchOk.ml:
```diff
@@ -453,9 +453,7 @@ let searching_fields conf base =
-                (if acc = "" then v
-                else if v = "" then acc
-                else acc ^ " / " ^ v)
+                (if acc = "" then v else if v = "" then acc else acc ^ " / " ^ v)
```
possibly!

swrup commented on test/test_utils.ml:
```diff
@@ -214,5 +264,6 @@ let suite =
            "util_string_with_macros" >:: util_string_with_macros;
            "util_transl_a_of_b" >:: util_transl_a_of_b;
+           "util_eval_translate" >:: util_eval_translate;
          ];
```
thanks for adding tests!!

for tests maybe the alcotest PR which is mergeable now should be merged before those new test, you will have to change some stuff but it should not be too hard

swrup replied:
also if you manage to write test for Cousins it would be nice

hgouraud committed:
corrections after swurp comments on PR

hgouraud committed:
update tests for alcotest. use 10/10/10 in remainder

hgouraud committed:
read_or_create_value for cousins table and limit to v=

hgouraud committed:
fix renaming of individual in caroussel

issue #1624

hgouraud committed:
particles management in books of surnames with improved anchors

* Particles are now managed in the book of surname and particles.txt has been
completed.
* Particles present in that file are written at the end of entries in parentheses
but the normal form of a surname is restored in the input itself before renaming.
* The anchor system to go back to the edited entry is fixed when renaming to a
first letters with accentuated characters. If the first letter is changed, two
links are given to fallback either to the new modified entry or to the previous
letter (to continue editing there).
* The anchor system now uses the first two letters of the entries for s=.. fallback.
This allows to bypass the number of entries limitation more often and to fallback
more frequently on the right modified entry.
* Renaming of unknown surnames and first_names (sn=? or fn=?) now don’t rename
unknowns  individuals "? ?" at all.

hgouraud committed:
empty %link_next; when no more shortest path found

issue #1628. needs side effects verification

a2line committed:
missing space

a2line committed:
missing name when deleting

a2line committed:
add second button to restore previous Sosa 1

use extra i/pnoczB variables to store previous sosa 1 and restore it
add same button that those in menubar
simplify implx button (still wip to hide implexes)

hgouraud committed:
menubar, use url_set rather that foreach;env_b for sosa navigation

hgouraud committed:
provide full title for updFam/delete

michelmno committed:
Handle properly GW_STATIC_PATH in gwd.ml

Required as correction of issue #1330

* the images subdirectory is at same level as etc subdirectory
  so image_prefix set accordingly.
* Note that the usage of "String.concat Filename.dir_sep"
  let's assume the provided GW_STATIC_PATH do not have trailing
  dir separator to avoid double separator after concat.
* From the above assumption of no trailing separator for GW_STATIC_PATH
  need to add a single separator for setting of static_path
  as expected in all usage of %static_path; in template files.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
manage utf-8 vowels

michelmno committed:
Replace %bvar.static_path; by %static_path; (step3)

* This is a complement of two commits:
  f4c4e1b7c Replace %bvar.static_path; by %static_path; in index.txt
  f18f92a06 Replace %bvar.static_path; by %static_path; (step2)

* that one concern carrousel.txt and doc_templm.txt

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Do not overwrite static_path by bvar if still present

This is to complete the previous code changes
about bvar.static_path superseded by GW_STATIC_PATH

To cover following specific use-case:
Use same database to test on same machine
different versions of geneweb triggered by different cgi scripts,
among which geneweb 7.0.0 that still need static_path from gwf,
but the other newer versions using GW_STATIC_PATH from cgi scripts.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

a2line committed:
fix CSS htree rendering

rename tree to h6, add pm_ prefix to pmod buttons to avoid conflicts
fix b.css if undefined

michelmno committed:
Avoid Uncaught TypeError: home_errors is null in welcome page

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
suppress unwanted space at head of set/get_var

hgouraud committed:
define url_setx with n evar to be zapped

a2line committed:
revamp all pmod scripts with highlight buttons and new alert system for failure

* manage highlights of buttons of selected modules for ALL jquery functions!
* transfer translations to lexicon for convenience.
* add a new alert system pop if the typed module is not existing, it is
dismissable by clicking on the alert itself (no close button).
* factorise and optimise the code with a lot of comments.
* add and load minified version p_mod.min.js (gain nearly 50 % on file size)

hgouraud committed:
find sosa_ref in templ.ml

hgouraud committed:
fix sosa.v in perso_utils.txt

hgouraud committed:
fix in gwd (errors_undef); get_var/set_var in dagDisplay

hgouraud committed:
fix broken url_set2; p_of_index to return '' if hidden

hgouraud committed:
updates to relations, sources and unions modules as suggested by D Deram

hgouraud committed:
-auth argument incompatible with CGI mode

a2line committed:
hide empty variables, remove superflous quote in title

hgouraud committed:
uncomment else part in %if(b.welcome_logo)
